astropy/coordinates/builtin_frames/intermediate_rotation_transforms.py

```diff
@@ -244,32 +243,38 @@ def precessedgeo_to_gcrs(from_coo, to_frame):
     pmat = gcrs_precession_mat(from_coo.equinox)
     crepr = from_coo.cartesian.transform(matrix_transpose(pmat))
     gcrs_coo = GCRS(crepr,
-                    obstime=to_frame.obstime,
-                    obsgeoloc=to_frame.obsgeoloc,
-                    obsgeovel=to_frame.obsgeovel)
+                    obstime=from_coo.obstime,
+                    obsgeoloc=from_coo.obsgeoloc,
+                    obsgeovel=from_coo.obsgeovel)
 
     # then move to the GCRS that's actually desired
     return gcrs_coo.transform_to(to_frame)
 
 
 @frame_transform_graph.transform(FunctionTransformWithFiniteDifference, TEME, ITRS)
 def teme_to_itrs(teme_coo, itrs_frame):
-    # first get us to TEME at the target obstime
-    # TODO: self transform?
-    teme_coo2 = teme_coo.transform_to(TEME(obstime=itrs_frame.obstime))
+    # use the pmatrix to transform to ITRS in the source obstime
```

Comment:
This change looks like it should not affect any results, right?

Reply:
This change affects results.  In the current code, where no `TEME` loopback is actually defined, that transformation line doesn't actually do anything, even though it looks like it should.  Thus, the current results are incorrect for a change in `obstime`.

I've extracted this fix as #12152, along with the `PrecessedGeocentric`->`GCRS` bugfix.